rust/src/nosetuid.rs

```diff
@@ -0,0 +1,45 @@
+//! Implements the `nosetuid` treefile option which removes SetUID and SetGID
+//! bits from all executables.
+
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use fn_error_context::context;
+use walkdir::WalkDir;
+use std::fs::File;
+use std::os::unix::fs::PermissionsExt;
+
+use crate::cxxrsutil::CxxResult;
+use crate::ffiutil::*;
+
+const Executable: u32 = 0o0111;
+const SetUID: u32 = 0o4000;
+const SetGID: u32 = 0o2000;
+
+/// Remove SetUID and SetGID bits from all executables.
+#[context("Removing SetUID & SetGID bits from all executables")]
+fn remove_setuid(_rootfs_dfd: &openat::Dir) -> Result<()> {
+    for e in WalkDir::new("usr") {
```

Comment:
So doesn't this actually operate on the process cwd?  Which...I don't think we necessarily make be the rootfs.

I think while this `walkdir` crate is obviously nice code, today it isn't integrating with `cap-std`, and I've been trying to push things towards that.  I am not sure that `walkdir` is buying us much over a simple recursive traversal function which we have in a few other places.

Reply:
xref https://github.com/bytecodealliance/cap-std/issues/82#issue-667453294